app/src/main/java/com/android/sample/ui/profile/Profile.kt

```diff
@@ -57,12 +57,25 @@ fun ProfileScreen(
     navigationActions: NavigationActions,
     listActivitiesViewModel: ListActivitiesViewModel
 ) {
-  val profileState = userProfileViewModel.userState.collectAsState()
```

Comment:
added offline support by integrating the NetworkManager to check network availability and fetch cached profile data from Room when offline, ensuring seamless functionality in both online and offline modes.